mirascope/core/mistral/_utils/_setup_call.py

```diff
@@ -95,28 +94,28 @@ def setup_call(
     messages = cast(list[BaseMessageParam | ChatMessage], messages)
     messages = convert_message_params(messages)
     if json_mode:
-        call_kwargs["response_format"] = ResponseFormat(
-            type=ResponseFormats("json_object")
-        )
+        call_kwargs["response_format"] = ResponseFormat(type="json_object")
         json_mode_content = _utils.json_mode_content(
             tool_types[0] if tool_types else None
         )
         if messages[-1].role == "user":
             messages[-1].content += json_mode_content
         else:
-            messages.append(ChatMessage(role="user", content=json_mode_content.strip()))
+            messages.append(
+                make_chat_message(role="user", content=json_mode_content.strip())
+            )
         call_kwargs.pop("tools", None)
     elif extract:
         assert tool_types, "At least one tool must be provided for extraction."
-        call_kwargs["tool_choice"] = cast(ToolChoice, ToolChoice.any)
+        call_kwargs["tool_choice"] = cast(ToolChoiceEnum, "any")
     call_kwargs |= {"model": model, "messages": messages}
 
     if client is None:
-        client = (
-            MistralAsyncClient() if inspect.iscoroutinefunction(fn) else MistralClient()
+        client = Mistral(api_key=os.environ.get("MISTRAL_API_KEY"))
+    if inspect.iscoroutinefunction(fn):
```

Comment:
you should use our internal helper utility [`fn_is_async`](https://github.com/Mirascope/mirascope/blob/7c1f6e3eae4aa086f5ee9685884516b75820af08/mirascope/core/base/_utils/_protocols.py#L110) for better typing here.

If it's not clear because of where the function is located, feel free to pull `fn_is_async` out into it's own file like other utilities.